src/wazuh_modules/syscollector/ports/portBSDWrapper.h

```diff
@@ -0,0 +1,158 @@
+        if (it != PORTS_TYPE.end())
+        {
+            retVal = AF_INET6 == m_socketInfo->psi.soi_family ? it->second + "6" : it->second;
+        }
+        return  retVal;
+    }
+    std::string localIp() const override
+    {
+        char ipAddress[NI_MAXHOST] = { 0 };
+        if(AF_INET6 == m_socketInfo->psi.soi_family)
+        {
+            sockaddr_in6 socketAddressIn6;
+            socketAddressIn6.sin6_family = m_socketInfo->psi.soi_family;
+            socketAddressIn6.sin6_addr = static_cast<in6_addr>(m_socketInfo->psi.soi_proto.pri_in.insi_laddr.ina_6);
+            getnameinfo(reinterpret_cast<sockaddr *>(&socketAddressIn6), sizeof(socketAddressIn6), ipAddress, sizeof(ipAddress), nullptr, 0, NI_NUMERICHOST);
+        }
+        else
+        {
+            sockaddr_in socketAddressIn;
+            socketAddressIn.sin_family = m_socketInfo->psi.soi_family;
+            socketAddressIn.sin_addr = static_cast<in_addr>(m_socketInfo->psi.soi_proto.pri_in.insi_laddr.ina_46.i46a_addr4);
+            getnameinfo(reinterpret_cast<sockaddr *>(&socketAddressIn), sizeof(socketAddressIn), ipAddress, sizeof(ipAddress), nullptr, 0, NI_NUMERICHOST);
+        }
+        return ipAddress;
+    }
+    int32_t localPort() const override
+    {
+        return ntohs(m_socketInfo->psi.soi_proto.pri_in.insi_lport);
+    }
+    std::string remoteIP() const override
+    {
+        char ipAddress[NI_MAXHOST] = { 0 };
+        if(AF_INET6 == m_socketInfo->psi.soi_family)
+        {
+            sockaddr_in6 socketAddressIn6;
+            socketAddressIn6.sin6_family = m_socketInfo->psi.soi_family;
+            socketAddressIn6.sin6_addr = static_cast<in6_addr>(m_socketInfo->psi.soi_proto.pri_in.insi_faddr.ina_6);
+            getnameinfo(reinterpret_cast<sockaddr *>(&socketAddressIn6), sizeof(socketAddressIn6), ipAddress, sizeof(ipAddress), nullptr, 0, NI_NUMERICHOST);
+        }
+        else
+        {
+            sockaddr_in socketAddressIn;
+            socketAddressIn.sin_family = m_socketInfo->psi.soi_family;
+            socketAddressIn.sin_addr = static_cast<in_addr>(m_socketInfo->psi.soi_proto.pri_in.insi_faddr.ina_46.i46a_addr4);
+            getnameinfo(reinterpret_cast<sockaddr *>(&socketAddressIn), sizeof(socketAddressIn), ipAddress, sizeof(ipAddress), nullptr, 0, NI_NUMERICHOST);
+        }
+        return ipAddress;
+    }
+    int32_t remotePort() const override
+    {
+        return ntohs(m_socketInfo->psi.soi_proto.pri_in.insi_fport);
+    }
+    int32_t txQueue() const override
+    {
+        return 0;
+    }
+    int32_t rxQueue() const override
+    {
+        return 0;
+    }
+    int32_t inode() const override
+    {
+        return 0;
+    }
+    std::string state() const override
+    {
+        std::string retVal { "unknown" };
+
+        const auto itProtocol { PORTS_TYPE.find(m_socketInfo->psi.soi_kind) };
+
+        if (PORTS_TYPE.end() != itProtocol && SOCKINFO_TCP == itProtocol->first)
```

Comment:
you can check m_socketInfo->psi.soi_kind == SOCKINFO_TCP at the beginning of the function and if true then access the map with .at() directly.
You can avoid the search on PORTS_TYPE map.

Reply:
what it indicates is true, but in this way I make sure that this value is in the reference map, for the correct translation.